Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/qrcode/QrCodeGenerator.kt

```diff
@@ -0,0 +1,65 @@
+package de.rki.coronawarnapp.ui.eventregistration.organizer.details
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Color
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.EncodeHintType
+import com.google.zxing.MultiFormatWriter
+import com.google.zxing.common.BitMatrix
+import dagger.Reusable
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import timber.log.Timber
+import javax.inject.Inject
+
+@Reusable
+class QrCodeGenerator @Inject constructor(
+    private val appConfigProvider: AppConfigProvider,
+    @AppContext private val context: Context,
+) {
+
+    private val errorMessage = SingleLiveEvent<String>()
```

Comment:
Since this is not  attached to any fragment and ideally should be used from ViewModel , maybe you change how you report errors such as throw an exception if the bitmap could not be generated.

```kotlin
@Throws(ExceptionTypeHere)
suspend fun createQRCode(input: String, size: Int = 1000){
   // Same code without try catch block , so the caller  of this function can handle the exception
}

Reply:
Valid point, will change it